### ISSUE TITLE
Options not Correctly Passed to Mocha Process?

### ISSUE THREAD

**Issue by benmarten** [grawk/grunt-loop-mocha#21]
Hi Matt,

I try to get Mocha to not stop when running via grunt-loop-mocha. I tried adding: `bail: false` to options/mocha. This does actually work when running mocha directly...

Now this what i see in console:
`[grunt-loop-mocha] mocha argv:  --globals,should,--timeout,120000,--slow,60000,--ignoreLeaks,--ui,bdd,--reporter,spec,--bail,test/index.js,test/nemo-test.js`

Maybe it should rather be: `...,--bail,false,...` ?

Basically my question is: How to get grunt-loop-mocha to not stop when a test step fails?

**Comment by grawk** [comment]
Try bail: "false"

**Comment by benmarten** [comment]
No, unfortunately doesn't work...

`Error: cannot resolve path (or pattern) 'false'`

Any other ideas how to not stop... ?

**Comment by grawk** [comment]
I think it's a legit bug. Will have to look into it...

**Comment by grawk** [comment]
loopmocha has a "noFail" property. does that do what you're looking for?

**Comment by grawk** [comment]
I think this line is the problem here:
https://github.com/grawk/grunt-loop-mocha/blob/master/tasks/loop-mocha.js#L103

the condition probably is true for 0 or false.

we can either look specifically for the "bail" key and make sure the value is set, or else modify this condition to identify zero versus boolean false.

**Comment by grawk** [comment]
Hey @benmarten after checking the docs I see that:
* the default behavior of mocha is to NOT bail based on a single test failure.
* the `--bail` arg doesn't require a value, it simply changes the default behavior
* if you provide the bail arg as `bail: ''` then loopmocha passes it onto mocha

**Comment by grawk** [comment]
grunt-loop-mocha will not stop if a test fails. Perhaps you are running into an uncaught exception scenario that crashes the overall process. The only option there would be to catch the exception or avoid it.

**Comment by benmarten** [comment]
ok, thanks for your help. i'll double check what's going on... maybe setup a sample project if i can't solve it...

**Comment by benmarten** [comment]
Yes, exactly that was it... Basically we were throwing an exception, hence the process was stopped.

Now handling it this way: http://www.adaltas.com/blog/2013/03/27/test-uncaughtException-error-mocha/

Thanks!
